consoles/VNC.pm

```diff
@@ -131,14 +131,23 @@ sub login {
     my $endtime = time + $connect_timeout;
 
     my $socket;
+    my $err_cnt = 0;
     while (!$socket) {
         $socket = IO::Socket::INET->new(
             PeerAddr => $hostname,
             PeerPort => $port,
             Proto    => 'tcp',
         );
         if (!$socket) {
+            $err_cnt++;
             return if (time > $endtime);
+            # we might be too fast trying to connect to the VNC host (e.g.
+            # qemu) so ignore the first three occurences of a failed
+            # connection attempt.
+            if ($err_cnt < 3) {
```

Comment:
you should just  write 
if ($err_cnt < 3) {
bmwqemu::diag "Error connecting to host <$hostname>: $@";
}

and remove duplicated lines 
sleep 1;
next;